batch.go

```diff
@@ -991,3 +1004,103 @@ func (batch *Batch) Equal(other Batcher) bool {
 	}
 	return len(batch.Entries) == equalEntries && equalEntries != 0
 }
+
+// WithOffset sets the Offset information onto a Batch so that during Create a "balanced offset"
+// record is appended at the end.
+//
+// Balanced offset records sum a File's Debits or Credits to zero in a single transaction.
+// Offset files are used to offset transactions from a single account inside of the ODFI.
+//
+// If the Batch contains Debits the offset record will be a Credit and if the batch contains Credits
+// the offset record will be a debit.
+func (b *Batch) WithOffset(off *Offset) {
+	b.offset = off
+}
+
+func (b *Batch) upsertOffset() error {
+	if b == nil || b.offset == nil {
+		return nil
+	}
+
+	// remove any Offset records already on the batch
+	addOffset := true // append to EntryAddendaCount
+	for i := 0; i < len(b.Entries); i++ {
+		// TODO(adam): Should we remove this based on checking the last element is
+		// debit/credit and sums to all the other elements (which are mutually exclusive to
+		// the last record being debit or credit)?
+		if strings.EqualFold(b.Entries[i].IndividualName, "OFFSET") {
+			b.Entries = append(b.Entries[:i], b.Entries[i+i:]...)
+			addOffset = false // we're replacing an EntryDetail, not adding
+			i++
+		}
+	}
+	if b.Control.TotalDebitEntryDollarAmount > 0 && b.Control.TotalCreditEntryDollarAmount > 0 {
+		return fmt.Errorf("WithOffset: batch must contain only debits=%d or credits=%d", b.Control.TotalDebitEntryDollarAmount, b.Control.TotalCreditEntryDollarAmount)
+	}
+	if utf8.RuneCountInString(b.offset.RoutingNumber) != 9 {
+		return fmt.Errorf("WithOffset: invalid routing number %s", b.offset.RoutingNumber)
+	}
+
+	ed := NewEntryDetail()
+	switch b.offset.AccountType {
+	case OffsetChecking:
+		if b.Control.TotalDebitEntryDollarAmount > 0 {
+			ed.TransactionCode = CheckingCredit
+		} else {
+			ed.TransactionCode = CheckingDebit
+		}
+	case OffsetSavings:
+		if b.Control.TotalDebitEntryDollarAmount > 0 {
+			ed.TransactionCode = SavingsCredit
+		} else {
+			ed.TransactionCode = SavingsDebit
+		}
+	default:
+		return fmt.Errorf("offset: unknown account type %v", b.offset.AccountType)
+	}
+	ed.RDFIIdentification = b.offset.RoutingNumber[:8]
+	ed.CheckDigit = b.offset.RoutingNumber[8:9]
+	ed.DFIAccountNumber = b.offset.AccountNumber
+	ed.IdentificationNumber = "" // TODO(adam): what to put here?
```

Comment:
What should go here? Any ideas?

Reply:
I don't see this being used for this use case. 
"An internal identification (alphanumeric) that you use to uniquely identify this Entry Detail Record"

Reply:
I'll leave it as an empty string then.